src/data/repositories/TeamMemberD2Repository.ts

```diff
@@ -15,10 +15,12 @@ export class TeamMemberD2Repository implements TeamMemberRepository {
                     fields: d2UserFields,
                 },
             })
-        ).map(response => {
-            if (!response.users) throw new Error("Team Members not found");
-            return response.users.map(d2User => this.mapUserToTeamMember(d2User));
-        });
+        )
+            .flatMap(response => assertOrError(response.users, `Team Members not found`))
+            .map(d2Users => {
+                if (!d2Users) throw new Error("Team Members not found");
```

Comment:
d2Users is an array, so this is always false. You want to check if empty? 

Also, we should return a Future.error (then flatMap instead of map)